hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/util/RestoreTool.java

```diff
@@ -487,12 +489,21 @@ private void checkAndCreateTable(Connection conn, Path tableBackupPath, TableNam
       if (createNew) {
         LOG.info("Creating target table '" + targetTableName + "'");
         byte[][] keys;
-        if (regionDirList == null || regionDirList.size() == 0) {
-          admin.createTable(htd);
-        } else {
-          keys = generateBoundaryKeys(regionDirList);
-          // create table using table descriptor and region boundaries
-          admin.createTable(htd, keys);
+        try{
+          if (regionDirList == null || regionDirList.size() == 0) {
+            admin.createTable(htd);
+          } else {
+            keys = generateBoundaryKeys(regionDirList);
+            // create table using table descriptor and region boundaries
+            admin.createTable(htd, keys);
+          }
+        } catch (NamespaceNotFoundException e){
+          LOG.warn(e.getMessage());
+          String namespaceAsString = targetTableName.getNamespaceAsString();
+          LOG.info("Creating target  namespace '" + namespaceAsString + "'");
+          admin.createNamespace(NamespaceDescriptor.create(namespaceAsString).build());
+          checkAndCreateTable(conn, tableBackupPath, tableName, targetTableName, regionDirList, htd,
```

Comment:
Why we call this `checkAndCreateTable` here, instead of doing same call we currently do in line 494 or 498?

Reply:
Thank you for your advice.